### PR TITLE
DrivAerML: slices reduction for more epochs (64, 48, 32)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

DrivAerML currently only gets 2 epochs in the 30-minute timeout at default slices=96. This is the same bottleneck that plagued TandemFoil — where reducing to slices=64 enabled 11 epochs and produced a -42% improvement. DrivAerML epoch time is ~10-11 min/epoch at slices=96. At slices=64, we should get 4-5 epochs; at slices=48 perhaps 6-7; at slices=32 perhaps 8-10. More training should significantly close the gap toward the 3.71% target.

## Instructions

Run sequentially (ONE AT A TIME). Use the winning config: no-EMA, AdamW lr=8e-4, T_max=150.

```bash
cd target/icml2026

# Run 1: slices=64, no-EMA, AdamW lr=8e-4
python train.py --dataset drivaerml --optimizer adamw --lr 8e-4 \
  --cosine_t_max 150 --no-use-ema --model_slices 64 \
  --wandb_group thorfinn-drivaerml-slices --wandb_name thorfinn-drivaerml-s64

# Run 2: slices=48, no-EMA, AdamW lr=8e-4
python train.py --dataset drivaerml --optimizer adamw --lr 8e-4 \
  --cosine_t_max 150 --no-use-ema --model_slices 48 \
  --wandb_group thorfinn-drivaerml-slices --wandb_name thorfinn-drivaerml-s48

# Run 3: slices=32, no-EMA, AdamW lr=8e-4
python train.py --dataset drivaerml --optimizer adamw --lr 8e-4 \
  --cosine_t_max 150 --no-use-ema --model_slices 32 \
  --wandb_group thorfinn-drivaerml-slices --wandb_name thorfinn-drivaerml-s32
```

**Report** `val_primary/surface_rel_l2_pct`, test metric, epochs completed for each run, and per-epoch loss trajectory. Key metrics: how many epochs did each slices setting get? Target: beat **56.91%**.

## Baseline

| Dataset | Metric | Value | PR |
|---|---|---|---|
| DrivAerML | val_primary/surface_rel_l2_pct | **56.91%** | #2467 (no-EMA, AdamW lr=8e-4, 2 epochs) |
| DrivAerML | test_primary/surface_rel_l2_pct | **57.33%** | #2467 |

Reproduce baseline: `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 8e-4 --cosine_t_max 150 --no-use-ema`

**Context:** External target is <3.71% (AB-UPT, ~500 epochs). Epoch count is the dominant lever — TandemFoil went from 197.87 → 114.92 (-42%) purely by getting 11 epochs vs 2. We need to find if a similar gain exists for DrivAerML.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT thorfinn:

**OOM issue encountered.** DrivAerML at slices=64, batch_size=1 (and batch_size=2) OOMs on a significant fraction of training samples (~31/394 = 8%). The largest DrivAerML meshes exceed 95 GB GPU memory during forward+backward pass. Added OOM batch-skipping to train_one_epoch but skipping 8% of training data is unacceptable.

**Fix:** Using `--drivaerml-train-surface-points 80000 --drivaerml-eval-surface-points 80000` to cap surface points per sample. This was also used in baseline PR #2467 retry runs. Will still use `--batch-size 1` for safety. Running all 3 slice configs (64, 48, 32) sequentially with this fix.

Timestamp: 2026-04-20 ~21:15 UTC

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2487 because Superseded: thorfinn reassigned to #2560 (MQA audit).
